generator/src/main/java/com/scottlogic/deg/generator/ConfigSource.java

```diff
@@ -0,0 +1,18 @@
+package com.scottlogic.deg.generator;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * Parent for command line configuration options.
+ */
+public interface ConfigSource {
+
+    boolean dontOptimise();
+    Path getOutputPath();
+    File getProfileFile();
+    boolean getValidateProfile();
+    boolean overwriteOutputFiles();
+    boolean shouldViolate();
```

Comment:
I dont think `shouldViolate` is relevant to visualise, should this be removed from this interface?

Reply:
moved into `GenerateConfigSource`.